File: Baekjoon/Class2/26.2164 카드2.java
import java.io.*;
import java.util.ArrayDeque;
import java.util.Deque;

class Main {
    public static void main(String[] args) throws IOException {
        int n;
        Deque<Integer> cards = new ArrayDeque<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(br.readLine());
        for(int i = 1; i <= n; i++)
            cards.addLast(i);

        while(cards.size() > 1) {
            cards.removeFirst(); // 맨 위 제거
            cards.addLast(cards.removeFirst()); // 맨 위 -> 맨 아래
        }
        System.out.print(cards.peekFirst());
        br.close();
    }
}

File: Baekjoon/Class2/26.2164 카드2.py
from collections import deque
import sys
input = sys.stdin.readline

n = int(input())
cards = deque(range(1, n + 1))

while len(cards) > 1:
    cards.popleft() #맨 위 제거
    cards.append(cards.popleft()) #맨 위 -> 맨 아래
print(cards[0])